.tekton/.currency/currency-tasks.yaml

```diff
@@ -0,0 +1,89 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: git-clone-task
+spec:
+  params:
+  - name: revision
+    type: string
+  workspaces:
+    - name: task-pvc
+      mountPath: /workspace
+  steps:
+    - name: clone-repo
+      # alpine/git:2.43.0
+      image: alpine/git@sha256:6ff4de047dcc8f0c7d75d2efff63fbc189e87d2f458305f2cc8f165ff83309cf
+      script: |
+        #!/bin/sh
+        echo "Cloning repo"
+        cd /workspace && git clone --filter=blob:none --sparse --depth 1 https://github.com/instana/python-sensor -b $(params.revision)
+        cd python-sensor
+        git sparse-checkout add .tekton/.currency
+        ls -lah /workspace/python-sensor
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: generate-currency-report-task
+spec:
+  workspaces:
+    - name: task-pvc
+      mountPath: /workspace
+  steps:
+    - name: generate-currency-report
+      # 3.10.13-bookworm
+      image: python@sha256:c970ff53939772f47b0672e380328afb50d8fd1c0568ed4f82c22effc54244fc
+      script: |
+        #!/usr/bin/env bash
+        cd /workspace/python-sensor/.tekton/.currency
+
+        python -m venv /tmp/venv
+        source /tmp/venv/bin/activate
+        pip install -r resources/requirements.txt
+
+        python scripts/generate_report.py
+        cat docs/report.md
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: upload-currency-report-task
+spec:
+  params:
+    - name: github-token-secret
+      default: instanacd-github-api-token
+  workspaces:
+    - name: task-pvc
+      mountPath: /workspace
+  steps:
+    - name: upload-currency-report
+      # alpine/git:2.43.0
+      image: alpine/git@sha256:6ff4de047dcc8f0c7d75d2efff63fbc189e87d2f458305f2cc8f165ff83309cf
+      env:
+        - name: GH_ENTERPRISE_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: $(params.github-token-secret)
+              key: "GH_ENTERPRISE_TOKEN"
+      script: |
+        #!/bin/sh
+
+        cd /workspace        
+        git clone https://oauth2:$GH_ENTERPRISE_TOKEN@github.ibm.com/instana/tracer-reports.git
+
+        if [ $? -ne 0 ]; then
+            echo "The attempt to clone the tracer-reports repository failed, preventing the upload of python tracer currency report." >&2
+            exit 1
+        fi
+        
+        cd tracer-reports
+
+        cp ../python-sensor/.tekton/.currency/docs/report.md ./automated/currency/python/report.md
```

Comment:
This unconditional overwriting can cause us serious problems and alarms, when it keeps downgrading things that it shouldn't. https://instana.slack.com/archives/C046513TNJ3/p1716390471672299

Reply:
Hi @Ferenc- , according to my understanding the downgrading wouldn't happen for any other libraries than `net-http` because [this](https://github.com/instana/ruby-sensor/blob/a390ed75d3f8cb75dff8f27315e915a5c2a3e696/.tekton/.currency/scripts/generate_report.py#L14) hardcoded entry is used only by `net-http` at the moment and will be removed when we install the gem explicitly. Please do let me know if you have any different thoughts on this.

Reply:
The way I see downgrading happens because of the combination of two reasons:
1. We generally allow downgrading here by allowing the overwriting of higher values with smaller ones in the report.
2. The generate report (has some coding issue similar to the forgotten hardcoded value) or simply doesn't run in the right environment (where the right dependencies are installed) because of outdated / faulty CI code.

Since we have seen occurences of the `2.`  condition and it is not entirely excludable, the idea is to eliminate the `1.` condition, so in case `2.` happens then the team doesn't get Jira issues, and pings from management etc.
Now, as eliminating `1.` might be a bigger effor, it doesn't have to be part of this PR, so this PR could go ahead from my side.

Reply:
Thank you for the explanation Ferenc, I will keep that in mind.